.github/workflows/ci.yml

```diff
@@ -55,42 +55,25 @@ jobs:
       - name: Install go-junit-report
         run: go install github.com/jstemmer/go-junit-report/v2@latest
 
+      - name: Set report filename
+        run: echo "REPORT_FILENAME=$(date +'%Y%m%d%H%M')_linodego_test_report.xml" >> $GITHUB_ENV
+
       - name: Run tests and save test report
         run: |
-          timestamp=$(date +'%Y%m%d%H%M')
-          report_filename="${timestamp}_linodego_test_report.xml"
-
-          if ! make test | go-junit-report -set-exit-code > "$report_filename"; then
-            echo "EXIT_STATUS=1" >> $GITHUB_ENV
-          fi
-          cat "$report_filename"
+          make test | go-junit-report -set-exit-code -iocopy -out $REPORT_FILENAME
         env:
           SKIP_LINT: 1
 
-      - name: Add additional information to XML report
-        if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+      - name: Upload test results to bucket
+        if: github.ref == 'refs/heads/main' && github.event_name == 'push' && always()
```

Comment:
The secrets of this repo is only available to `main` branch, hence above checks are required to get the workflow to pass on open PRs against main